icu4j/main/core/src/test/java/com/ibm/icu/dev/test/message2/TestUtils.java

```diff
@@ -139,5 +178,7 @@ private static Path getTestFile(Class<?> cls, String fileName) throws URISyntaxE
         URI getPath = cls.getClassLoader().getResource(packageName).toURI();
         Path filePath = Paths.get(getPath);
         Path json = Paths.get(fileName);
-        return filePath.resolve(json);
+        // Test files are in icu/testdata/
+        Path icuTestdataDir = filePath.resolve("../../../../../../../../../../../testdata/message2/").normalize();
+        return icuTestdataDir.resolve(json);
```

Comment:
Ideally should test and give the user some feedback if it doesn't work.  Also, see comment about packaging when using a non-repo work area.

Reply:
Done in [c53f203](https://github.com/unicode-org/icu/pull/3050/commits/c53f203999c18e1338de333bce3bd943167e5c87). Re: using a non-repo work area, see https://github.com/unicode-org/icu/pull/3050#issuecomment-2234510795